assignments/arrays.js

```diff
@@ -60,35 +60,69 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
 // for (let i = 0; i < arr.length; i++) {
 //     arr[i]; // 1,2,3,4
 // }
-
+console.log('hello');
 // ==== Challenge 1 ====
 // The dealer can't recall the information for a car with an id of 33 on his lot. Help the dealer find out which car has an id of 33 by logging the car's year, make, and model in the console log provided to you below:
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+for (let i = 0; i< inventory.length; i++){
+  if (inventory[i].id === 33)
+  {
+ console.log(`The car is ${inventory[i].car_year} ${inventory[i].car_make} ${inventory[i].car_model}`)
+}
+}
+
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
 let lastCar = 0;
-console.log();
+for (let i=0; i< inventory.length; i++)
+{
+lastCar = inventory.length;
+if (inventory[i].id === lastCar)
+{console.log(`The last car is ${inventory[i].car_make} ${inventory[i].car_model}`);}
+}
```

Comment:
I see your logic here, and it works. However, is it truly necessary to go through the entire array to get the last car object in the array?

The challenge is asking you just for the last object in the array. Notice that there are 50 objects in the array. Array indexing starts at 0, so the last object in the array would be at index 49 (_the 50th index_).

Therefore, you could simply say: `inventory[49]`

Notice that because of the zero-based indexing, the index of the last element of the array is one less than the index. Try playing with that a bit:

```js
const arrOf4Nums = [0, 1, 2, 3]; // notice that 3 is the last, which corresponds to index
const arrOf10Nums = [0, 1, 2, 3, 4, 5, 6, 7, 8 ,9]; // notice 9 is last
```

That means you could generalize the above to 

```js
inventory[inventory.length - 1];
```

That would get you the last element.

Reply:
Got it.I basically wanted to just make something that would work regardless of the size of the array, which is why I went for it